raiden/tests/integration/api/test_pythonapi_regression.py

```diff
@@ -44,10 +45,14 @@ def run_test_close_regression(raiden_network, deposit, token_addresses):
         registry_address, token_address, amount, api2.address, identifier=identifier
     )
     exception = ValueError("Waiting for transfer received success in the WAL timed out")
-    with gevent.Timeout(seconds=5, exception=exception):
-        waiting.wait_for_transfer_success(
-            app1.raiden, identifier, amount, app1.raiden.alarm.sleep_time
-        )
+    try:
+        with gevent.Timeout(seconds=5, exception=exception):
+            waiting.wait_for_transfer_success(
+                app1.raiden, identifier, amount, app1.raiden.alarm.sleep_time
+            )
+    finally:
+        for app in raiden_network:
```

Comment:
This would be better done inside `wait_for_transfer_success`, otherwise we have to duplicate it on every call site.

I think the best approach would be to rename the function to `wait_for_transfer_result`, and wait until either: `on-chain secret registartion`, `unlock`, or `expired`, then return the result (Please, add it in an Enum).

The tests would then assert on the expected result.